<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>reveal.js</title>

		<link rel="stylesheet" href="css/reveal.css">
		<link rel="stylesheet" href="css/theme/black.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="lib/css/zenburn.css">

		<!-- solarized theme -->
		<!-- <link rel="stylesheet" href="css/theme/solarized.css" id="theme"> -->

		<!-- google fonts -->
		<link href='http://fonts.googleapis.com/css?family=Source+Sans+Pro' rel='stylesheet' type='text/css'>

		<!-- fontawesome -->
		<!-- <link rel="stylesheet" href="css/fontawesome-free-5.0.13/web-fonts-with-css/css/fontawesome-all.css"> -->
		<script defer src="https://use.fontawesome.com/releases/v5.7.2/js/all.js" integrity="sha384-0pzryjIRos8mFBWMzSSZApWtPl/5++eIfzYmTgBBmXYdhvxPc+XcFEk+zJwDgWbP" crossorigin="anonymous"></script>

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>

		<style type="text/css">
			#rotate_text {
				-webkit-transform: rotate(-90deg);
				-moz-transform: rotate(-90deg);
				transform: rotate(-90deg);
				position: absolute;
				vertical-align: middle;
				position: absolute;
    		top: 500px;
    		left: -50px;
				filter:progid:DXImageTransform.Microsoft.BasicImage(rotation=3);
			}

			.reveal section ul.a {list-style-type: circle; margin: 0; padding: 0;}
			.reveal section ul.b {list-style-type: square; margin-left: 4; padding-left: 0;}

			/*.reveal section img {border: none;}*/

			.center-block {
			  display: block;
			  margin-left: auto;
			  margin-right: auto;
			}

			.reveal table th, td {
			  padding: 0px;
			}
		</style>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">
		
		<section id="intro" class="center">
			<h1><font color="#B8172F">Biodiversity &amp; Taxonomy Software Tools @ rOpenSci</font></h1><br>
			<p class="grey">Scott Chamberlain (<i class="fab fa-twitter"></i> <a href="https://twitter.com/sckottie">@sckottie</a>)
				<p class="grey">UC Berkeley / rOpenSci
			<br>
		</section>
	
		<!-- high level overview -->
		<section id="broad-areas" class="center">
			<h1>Broad areas of packages</h1>
			<ul class="b">
				<li><h2 align="left">Taxonomy</h2></li>
                <li><h2 align="left">Occurrence data</h2></li>
                <li><h2 align="left">Environmental data</h2></li>
			</ul>
		</section>

		<section id="citations">
			<h2>Citations</h2>
			<img src="img/citation.png" width="1000px">
		</section>

		<section id="questions">
			<h2>Questions addressed w/ our software</h2>
			<img src="img/questions.png" width="1050px">
		</section>

		<section>
			<section id="taxize">
				<h2>Use cases: taxize</h2>
				<ul class="b">
					<!-- <li>check sci. names against COL, get IUCN Red List status, in botanical inventory</li> -->
					<!-- <li>collect taxonomic information</li> -->
					<li>classify species invasive or not</li>
					<li>software uses taxize to check user names against ITIS</li>
					<li>check names against TPL, EOL, COL, IUCN, uBIO</li>
					<!-- <li>verify taxonomic classification (birds)</li>
					<li>verify taxonomic classification (plants)</li> -->
					<li>get name data for NCBI sequence data</li>
					<li>validate genus names for a food web</li>
					<li>compiled dataset of tropical forest tree species names checked w/ TNRS</li>
					<!-- <li>software depends on taxize to get taxon classifications</li> -->
					<li>add taxonomic classification data to meta-analysis dataset</li>
					<!-- <li>validate/clean names for plant-fungal association study</li> -->
				</ul>
			</section>
			<section id="rgbif">
				<h2>Use cases: rgbif</h2>
				<ul class="b">
					<!-- <li>occurrence counts globally through year range, comparing birds to other taxa</li> -->
					<!-- <li>occurrence records for mapping two species</li> -->
					<li>occurrence records to construct niche models</li>
					<li>collect occurrence records for catfishes in a Brazilian river</li>
					<li>occurrence records of Acacia species in Australia through time</li>
					<li>assessing niche expansion of invasive plants with occurrence records</li>
					<!-- <li>test of plant functional type classification with occurrences records</li> -->
					<!-- <li>use occurrence records with climate data to predict potential restoration plant species</li> -->
					<li>small note in manuscript about a species being in a study area</li>
					<!-- <li>occurrence records of trees in the Amazon to test predictors of seed size</li> -->
				</ul>
			</section>
			<section id="rfishbase">
				<h2>Use cases: rfishbase</h2>
				<ul class="b">
					<li>collect fish life history traits</li>
					<!-- <li>obtain species within a group, and two traits</li> -->
					<li>extract fecundity data for four fish species</li>
					<!-- <li>get family name for each species</li> -->
					<li>group species into trophic guilds using trophic position</li>
					<li>fetch salinity associated traits for many fish species</li>
					<!-- <li>calculate fish fractional size from max size for many fish species</li> -->
					<li>acquire depth ranges for many species to determine a phylogenetic signal</li>
					<!-- <li>calc. percentage of extant species living on reefs</li> -->
				</ul>
			</section>
			<section id="rentrez">
				<h2>Use cases: rentrez</h2>
				<ul class="b">
					<li>search PubMed for mentions of phrases</li>
					<!-- <li>search PubMed for mentions of phrases</li> -->
					<li>demonstrate rentrez use to search NCBI for articles in institutional repositories</li>
					<li>fetch NCBI taxonomic information for sequence data</li>
					<li>use NCBI's Gene Expression Omnibus service</li>
					<li>extract citations (presumably from PubMed) using rentrez</li>
				</ul>
			</section>
			<section id="spocc">
				<h2>Use cases: spocc</h2>
				<ul class="b">
					<li>use GBIF data to explore genome size variation against many variables</li>
					<!-- <li>grant proposal: data will be collected with spocc</li> -->
					<!-- <li>use GBIF data to construct species distributions</li> -->
					<li>use GBIF data to construct species range and niche centroids</li>
					<!-- <li>use GBIF, iDigBio, and BISON data to construct range maps</li> -->
					<li>use GBIF, VertNet, BISON, Ecoengine, iNaturalist data to construct species niche models</li>
					<li>use Vertnet and iNaturalist data to identify most vulernable populations for snakebites</li>
					<!-- <li>use GBIF data to map potential distributions of mosquito vectors based on ecological niche models</li> -->
					<li>use GBIF data via zoon in malariaAtlas R pkg</li>
					<!-- <li>use GBIF data to estimate climatic distributions of species</li> -->
					<!-- <li>use GBIF data to construct species distributions models</li> -->
					<li>use GBIF and iDigBio data to construct future species ranges</li>
				</ul>
			</section>
			<section id="rnoaa">
				<h2>Use cases: rnoaa</h2>
				<ul class="b">
					<li>fetch sea surface temperature (SST): check if latitude/SST explains variation in body size</li>
					<li>use many variables from ISD to predict airplane flight time</li>
					<li>use climate data to identify opportunities for stream restoration</li>
					<li>estimate interannual climatic variability in urban areas</li>
					<li>government report on precipitation</li>
				</ul>
			</section>
		</section>

		<section id="use-cases-rocitations" class="center">
			<h2><span style="font-size: 150px; color: Dodgerblue;"><i class="fab fa-twitter"></i></span><br>Follow <a href="https://twitter.com/rocitations">@rocitations</a> for use cases</h2>
			<br>
			<h2>Citations spreadsheet <a href="https://github.com/ropenscilabs/ropensci_citations/blob/master/citations.tsv">ropenscilabs/ropensci_citations => citations.tsv</a> for citations</h2>
			<br>
			<h2>Use cases spreadsheet <a href="https://github.com/ropenscilabs/ropensci_citations/blob/master/use_cases.tsv">ropenscilabs/ropensci_citations => use_cases.tsv</a> for use cases</h2>
		</section>

		<!-- taxonomy tools -->
		<section id="taxonomy-tools-intro">
			<h1>Taxonomy</h1>
			<br>
			<ul class="b">
				<li><h3 align="left">Taxonomic data: e.g., <i>Poa annua ssp. annua</i></h3></li>
				<br>
				<li><h3 align="left">There's lots of taxonomic data sources ...</h3></li>
				<li><h3 align="left">Data sources have diff. data ...</h3></li>
				<li><h3 align="left">&amp; diff. IDs for the same names ...</h3></li>
				<li><h3 align="left">Taxonomy data often not the goal ...</h3></li>
				<li><h3 align="left">But rather a necessary piece of research ...</h3></li>
			</ul>
		</section>

		<section id="taxonomy-tools-rundown">
			<h1>Taxonomy</h1>
			<ul class="b">
				<li><h3 align="left"><a href="https://github.com/ropensci/taxa">taxa</a> - Taxonomic classes and taxonomically aware data manipulation</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/taxize">taxize</a> - taxonomy data from many data sources</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/taxizedb">taxizedb</a> - taxize, but with local SQL databases</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/rentrez">rentrez</a> - NCBI data, including taxonomy data</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/worrms">worrms</a> - WORMS marine data, including taxonomy</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/ritis">ritis</a> - USGS's ITIS taxonomy</h3></li>
				<li><h3 align="left">... many others </h3></li>
			</ul>
			<br>
			<h3>R taxonomy task view: <a href="https://github.com/ropensci/taxonomy">github.com/ropensci/taxonomy</a></h3>
		</section>

		<!-- species occurrence tools -->
		<section id="occurrence-tools-intro">
			<h1>Occurrence data</h1>
			<br>
			<ul class="b">
				<li><h3 align="left">Occurrence data: e.g.:</h3></li>
				<pre>
key                    scientificName   decimalLatitude decimalLongitude
1986611609   Encelia californica Nutt.        32.66866        -117.0856
				</pre>
				<li><h3 align="left">Many data sources ...</h3></li>
				<li><h3 align="left">Data from collected/observed specimens ...</h3></li>
				<li><h3 align="left">Most feed (all or part of their data) into <a href="https://www.gbif.org/">GBIF</a> ...</h3></li>
			</ul>
		</section>

		<section id="occurrence-tools-rundown">
			<h1>Occurrence data</h1>
			<ul class="b">
				<li><h3 align="left"><a href="https://github.com/ropensci/rgbif">rgbif</a> - occurrence data from GBIF's > 1 billion records</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/rbison">rbison</a> - occurrence data from USGS's BISON</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/rvertnet">rvertnet</a> - occurrence data from VertNet (vertebrates)</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/rebird">rebird</a> &amp; <a href="https://github.com/CornellLabofOrnithology/auk">auk</a> - occurrence data from eBird</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/spocc">spocc</a> - occurrence data from many sources, single interface</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/finch">finch</a> - parse GBIF bulk data (Darwin Core)</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/CoordinateCleaner">CoordinateCleaner</a> & <a href="https://github.com/ropensci/scrubr">scrubr</a> - clean occurrence data</h3></li>
			</ul>
		</section>

		<!-- environmental data tools -->
		<section id="environmental-tools-intro">
			<h1>Environmental data</h1>
			<br>
			<ul class="b">
				<li><h3 align="left">Environmental data: e.g.:</h3></li>
				<pre>
station  	    variable    value
12334        precipitation_mm       34
				</pre>
				<li><h3 align="left">Tons of data sources ...</h3></li>
				<li><h3 align="left">Hierarchy of data: raw to various levels of cleaned/etc. ...</h3></li>
				<li><h3 align="left">Types: precipation, temperature, wind speed, humidity, snow fall, etc.</h3></li>
			</ul>
		</section>

		<section id="environmental-tools">
			<h1>Environmental data</h1>
			<ul class="b">
				<li><h3 align="left"><a href="https://github.com/ropensci/rnoaa">rnoaa</a> - download climate data (prec, temp, wind, storms, snow, etc.) from NOAA</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/rerddap">rerddap</a> - NOAA ERDDAP servers</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/riem">riem</a> - Iowa Environment Mesonet data</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/weathercan">weathercan</a> - Environment & Climate Change Canada</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/bomrang">bomrang</a> - Australian gov't Bureau of Meteorology</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/GSODR">GSODR</a> - Global Surface Summary of the Day</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/nasapower">nasapower</a> - NASA POWER data</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/hydroscoper">hydroscoper</a> - Greek National Data</h3></li>
				<!-- <li><h3 align="left"><a href="https://github.com/ropensci/clifro">clifro</a> - CliFlo data</h3></li> -->
				<!-- <li><h3 align="left"><a href="https://github.com/ropensci/getCRUCLdata">getCRUCLdata</a> - CRU CL v. 2.0 Climatology Elements</h3></li> -->
			</ul>
		</section>

		<!-- tools in development and other unsolved problems -->
		<section id="the-future" class="center">
			<h1>future work / <br> hard problems</h1>
		</section>

		<section id="taxonomy-tools-future">
			<h2>taxonomy tools: in the works</h2>
			<br>
			<ul>
				<li><h3>taxonomic name parsing: we need fast &amp; platform independent name parsing <a href="https://gitlab.com/gogna/gnparser/">gnparser</a>/<a href="https://github.com/ropenscilabs/pegax">pegax</a></h3></li>
				<!-- <li><h3>package <a href="https://github.com/ropenscilabs/taxadc">taxadc</a>: serialize R taxonomic data to Darwin Core- in early development</h3> </li> -->
				<li><h3><a href="https://github.com/ropensci/taxizedb">taxizedb</a> - hard to a) make similar interface to SQL DB's as web services &amp; 2) simplify varied database installs</h3></li>
				<ul>
					<li>see also: <a href="https://github.com/cboettig/taxadb">taxadb</a> &amp; Kari Norman's part</li>
				</ul>
				<li><h3><a href="https://github.com/ropensci/taxview">taxview</a> - summarise and visualize data sets WRT taxonomy</h3></li>
			</ul>
		</section>

		<section data-markdown id="taxview-example">
			<script type="text/template">
				## Summarise/visualize data sets by taxonomy - <a href="https://github.com/ropensci/taxview">taxview</a>
				```r
				library('taxview')
				x <- system.file("examples/plant_spp.csv", package = "taxview")
				```

				```r
				(dat <- tibble::as_tibble(
					 data.table::fread(x, stringsAsFactors = FALSE, 
					   data.table = FALSE)))
				#>  # A tibble: 130 x 2
				#>    name                        id
				#>    <chr>                    <int>
				#>  1 Dianthus engleri       1531994
				#>  2 Anacolosa frutescens   1618138
				#>  3 Hymenophyllum plicatum  638568
				(sumdat <- tv_summarise(dat_clean))
				#> <tv_summary>
				#>  no. taxa: 129
				#>  by rank: N (21)
				#>  by rank name: N (594)
				#>  within ranks: N (20)
				```
				```r
				sumdat$by_within_rank
				#> $superkingdom
				#> name      count percent
				#> 1 Eukaryota   127      98
				#> 2 Bacteria      2       2
				sumdat$by_within_rank$subphylum
				#>   name           count percent
				#> 1 Streptophytina   119      96
				#> 2 Craniata           2       2
				#> 3 Hexapoda           2       2
				#> 4 Chelicerata        1       1
				```
			</script>
		</section>

		<section data-markdown id="new-taxize-cli">
			<script type="text/template">
				## taxize: improved UI

				```r
				res <- get_gbifid(spp, messages = FALSE)
				══  10 queries  ══════════════
				✔  Found:  Diospyros cacharensis
				More than one GBIF ID found for taxon 'Piptocarpha riedelii'!

				            Enter rownumber of taxon (other inputs will return 'NA'):

				   gbifid                scientificname    rank   status matchtype
				1 5406610    Piptocarpha riedelii Baker species ACCEPTED     EXACT
				2 7526881 Piptocarpha riedelii Sch.Bip. species DOUBTFUL     EXACT
				1: 1
				Input accepted, took gbifid '5406610'.
				✔  Found:  Piptocarpha riedelii
				✔  Found:  Polystichum amboroensis
				✔  Found:  Schizymenium dolichothecum
				✔  Found:  Plicouratea granulosa
				✔  Found:  Eremurus anisopterus
				✖  Not Found:  Stapelia bifolia
				✔  Found:  Masdevallia gairiana
				✔  Found:  Nephrodium totta
				✔  Found:  Goniopteris rubrinervis
				══  Results  ═════════════════

				● Total: 10
				● Found: 9
				● Not Found: 1
				```
			</script>
		</section>

		<section data-markdown id="new-taxize-state">
			<script type="text/template">
				## taxize: maintaining state

				```r
				get_tsn(c(x, y z))
				# fails out after the 1st one
				#> Error ... 
				
				# then call another fxn to restart the process
				get_proceed() # by default uses last run get_* fxn
				#>  ...
				```

				what's currently working:


				```r
				taxon_last()
				#> <taxon state>
				#>  class: gbifid
				#>  elapsed (sec): 10.06
				#>  count: 10
				#>   Diospyros cacharensis: 4070227
				#>   Piptocarpha riedelii: 5406610
				#>   Polystichum amboroensis: 8360370
				#>   Schizymenium dolichothecum: 2677108
				#>   Plicouratea granulosa: 3696390
				#>   Eremurus anisopterus: 2778793
				#>   Stapelia bifolia: NA
				#>   Masdevallia gairiana: 3654347
				#>   Nephrodium totta: 7410470
				#>   Goniopteris rubrinervis: 4082465
				```
			</script>
		</section>

		<section data-markdown id="new-taxize-taxa-integration">
			<script type="text/template">
				## taxize: integrating taxa

				see [issue #508](https://github.com/ropensci/taxize/issues/508) and branch [taxa-ids](https://github.com/ropensci/taxize/tree/taxa-ids)

				```r
				splist <- c("annona cherimola", 'annona muricata', "quercus robur",
			       "shorea robusta", "pandanus patina", "oryza sativa", "durio zibethinus")
				res <- get_tsn(splist, messages=FALSE)
				#> <taxa>
				#>   no. taxa:  7
				#>   Annona cherimola / species / 506198
				#>   Annona muricata / species / 18098
				#>   Quercus robur / species / 19405
				#>   Shorea robusta / species / 506787
				#>   Pandanus patina / species / 507376
				#>   Oryza sativa / species / 41976
				#>   Durio zibethinus / species / 506099
				```
				<br>
				### Will facilitate all the cool stuff Zach is talking about next ...
			</script>
		</section>

		<section id="occurrence-tools-future">
			<h2>occurrence tools: in the works</h2>
			<ul>
				<li><h3>taking the pain out of GBIF downloads: <a href="https://github.com/ropensci/rgbif/issues/266">ropensci/rgbif#266</a>: queuing tool for GBIF downloads - <em>would love any feedback</em></h3></li>
				<li><h3><a href="https://github.com/ropensci/spocc/issues/210">auk integration</a> into spocc</h3></li>
				<li><h3>occurrence cleaning in R: <em>hard problem!</em> A few efforts:</h3></li>
				<ul>
					<li><h3>package <a href="https://github.com/ropensci/scrubr">scrubr</a></h3></li>
					<li><h3>package <a href="https://github.com/ropensci/onboarding/issues/210">CoordinateCleaner</a></h3></li>
					<li><h3>deduplication badly needed: aggregation up the data provider ladder</h3></li>
				</ul>
			</ul>
		</section>

		<!-- <section id="occurrence-tools-questions">
			<h2>occurrence tools: questions</h2>
			<br>
			<ul>
				<li><h3>DOIs for GBIF search service (and other services)?</h3></li>
				<li><h3>related to above: Sharing dataset associated with paper</h3></li>
				<li><h3>Visualzing huge occurrence datasets? (GBIF map API now in dev ver of <code>rgbif</code> ~ <a href="https://github.com/ropensci/rgbif/blob/master/R/map_fetch.R" target="_blank">rgbif::map_fetch</a>)</h3></li>
			</ul>
		</section> -->

		<section id="more-info">
			<h2>where to find out more</h2>
			<ul>
				<li><h3>taxonomy data: <a href="https://github.com/ropensci/taxonomy">taxonomy task view</a></h3></li>
				<li><h3>occurrence data:</h3></li>
				<ul>
					<li><h3><a href="https://ropensci.github.io/occurrence-manual/">in progress bookdown on occurrence data</a></h3></li>
					<li><h3><a href="https://peerj.com/preprints/3304/">rgbif/pygbif/gbifrb pre-print</a></h3></li>
				</ul>
				<li><h3>environmental data: <a href="https://ropensci.org/packages/">rOpenSci packages page</a></h3></li>
			</ul>
		</section>

		<!-- talk to us -->
		<section id="talk-to-us" class="center">
			<h1>talk to us <i class="fas fa-bullhorn"></i></h1>
			<br>
			<h2>discussion forum: <a href="https://discuss.ropensci.org/">discuss.ropensci.org</a></h2>
			<br><br>
			<h2>slides: <a href="https://scotttalks.info/rocctax">scotttalks.info/rocctax</a></h2>
			<!-- <h2>Made w/: <a href="https://github.com/hakimel/reveal.js">reveal.js v3.2.0</a>, <a href="http://getbootstrap.com/">Bootstrap v3.3.5</a>, <a href="http://fortawesome.github.io/Font-Awesome/">FontAwesome v5.0.13</a></h2> -->
		</section>
			</div>
		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.js"></script>

		<script>
			// More info about config & dependencies:
			// - https://github.com/hakimel/reveal.js#configuration
			// - https://github.com/hakimel/reveal.js#dependencies
			Reveal.initialize({
				history: true,
				center: false,
				width: 1400,
				height: 900,
				transition: "none",
				dependencies: [
					{ src: 'plugin/markdown/marked.js' },
					{ src: 'plugin/markdown/markdown.js' },
					{ src: 'plugin/notes/notes.js', async: true },
					{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } }
				]
			});
		</script>
	</body>
</html>
