<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>bocc</title>

		<meta name="description" content="bocc">
		<meta name="author" content="Scott Chamberlain">

		<meta name="apple-mobile-web-app-capable" content="yes">
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">

		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">

		<link rel="stylesheet" href="css/reveal.css">
		<link rel="stylesheet" href="css/theme/solarized.css" id="theme">
		<link rel="stylesheet" href="css/bootstrap/bootstrap.css">

		<!-- use google font -->
		<link href='http://fonts.googleapis.com/css?family=Source+Sans+Pro' rel='stylesheet' type='text/css'>
		<link rel="stylesheet" href="css/font-awesome/css/font-awesome.css">

		<!-- favicon -->
		<link rel="shortcut icon" type="image/x-icon" href="img/favicon.ico">

		<!-- icons -->
		<link rel="stylesheet" href="http://netdna.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.css">

		<!-- Code syntax highlighting -->
		<link rel="stylesheet" href="lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>

		<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>

		<style type="text/css">
			#rotate_text {
				-webkit-transform: rotate(-90deg);
				-moz-transform: rotate(-90deg);
				transform: rotate(-90deg);
				position: absolute;
				vertical-align: middle;
				position: absolute;
    		top: 500px;
    		left: -50px;
				filter:progid:DXImageTransform.Microsoft.BasicImage(rotation=3);
			}

			ul.a {list-style-type: circle; margin: 0; padding: 0;}
			ul.b {list-style-type: square; margin-left: 4; padding-left: 0;}

			img {border: none;}

			.center-block {
			  display: block;
			  margin-left: auto;
			  margin-right: auto;
			}
		</style>

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>
		<div class="reveal">
			<div class="slides">
		<section id="intro">
			<h1><font color="#B8172F">Biodiversity and Taxonomy Software Tools in R</font></h1><br>
			<p class="grey">Scott Chamberlain (<a href="https://twitter.com/sckottie">@sckottie</a>/<a href="https://twitter.com/ropensci">@ropensci</a>)
				<p class="grey">UC Berkeley / rOpenSci
			<br>
			<span><img src="img/icon_lettering_color.svg" alt="rOpenSci" width="800" /></span>
		</section>

		<section id="thanks-helmsley" class="center">
			<span><img src="img/helmsley.png" alt="hemlsley foundation" width="1000" /></span>
		</section>

		<section id="link" class="center">
			<i class="fa fa-external-link fa-5x"></i>
			<br><br>
			<h2><a href="https://scotttalks.info/bocc">scotttalks.info/bocc</a> </h2>
			<br><br>
			<h2><a href="https://scotttalks.info/bocc/slides_960-720.pdf">pdf slides 960x720</a></h2>
			<h2><a href="https://scotttalks.info/bocc/slides_1280-720.pdf">pdf slides 1280x720</a></h2>
		</section>

		<section id="license" class="center">
			<br>
			<h1>LICENSE: <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY 4.0</a></h1>
		</section>

		<section id="keyboard-shortcuts" class="center">
			<h2>Keyboard shortcuts: press <code>?</code></h2>
			<br>
			<img src="img/keyboard-shortcuts.png" width="600">
		</section>


		<!-- actual stuff -->
		<!-- about ropensci -->
		<section id="ropensci">
			<br>
			<span><img src="img/icon_lettering_color.svg" alt="rOpenSci" width="1200" /></span>
			<br>
			<center><a href="https://ropensci.org/">ropensci.org &nbsp;<i class="fa fa-external-link-square"></i></a></center>
		</section>

		<section id="ropensci-does">
			<h1>rOpenSci Does</h1>
			<br><br>
	    <i class="fa fa-users fa-5x" aria-hidden="true"></i>&nbsp;&nbsp;
	    <i class="fa fa-code fa-5x" aria-hidden="true"></i>&nbsp;&nbsp;
			<i class="fa fa-book fa-5x" aria-hidden="true"></i>&nbsp;&nbsp;
			<i class="fa fa-check-square-o fa-5x" aria-hidden="true"></i>&nbsp;&nbsp;
		</section>

		<section id="ropensci-staff">
			<h1>rOpenSci Staff</h1>
			<br>
			<a href="https://ropensci.org/about/#team">ropensci.org/about/#team</a>
			<br><br>
			<ul class="b">
				<li><h2 align="left">~5 full time</h2></li>
				<li><h2 align="left">leadership team</h2></li>
				<li><h2 align="left">advisory board</h2></li>
			</ul>
		</section>

	
		<!-- describe problem space -->
		<!-- FIXME: make slides -->
		
		<!-- discuss current tooling in R -->
		<!-- taxonomy tools -->
		<section id="taxonomy-tools">
			<h1>Taxonomy</h1>
			<ul class="b">
				<li><h3 align="left"><a href="https://github.com/ropensci/taxa">taxa</a> - Taxonomic classes and taxonomically aware data manipulation</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/taxize">taxize</a> - Taxonomic "toolbelt" - work w/ taxonomy web APIs</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/taxize">taxizedb</a> - taxize, but with local SQL databases</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/rentrez">rentrez</a> - NCBI's Entrez services</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/biomartr">biomartr</a> - Biomart R client</h3></li>
				<li><h3 align="left"><a href="https://bioconductor.org/packages/release/bioc/html/genbankr.html">genbankr</a> - Parse GenBank files into useful objects</h3></li>
				<li><h3 align="left"><a href="https://github.com/ropensci/rsnps">rsnps</a> - SNPs data retrieval</h3></li>
			</ul>
		</section>

		<section data-markdown id="taxize-example">
			<script type="text/template">
				## Taxonomic data from >20 sources - <a href="https://github.com/ropensci/taxize">taxize</a>
				<br>
				### Taxonomic hierarchies from NCBI/ITIS/COL/etc

				```r
				library('taxize')
				classification("Chironomus riparius", db = "gbif")
				```
				<br>
				```r
				#> $`Chironomus riparius`
				#>                  name    rank      id
				#> 1            Animalia kingdom       1
				#> 2          Arthropoda  phylum      54
				#> 3             Insecta   class     216
				#> 4             Diptera   order     811
				#> 5        Chironomidae  family    3343
				#> 6          Chironomus   genus 1448033
				#> 7 Chironomus riparius species 1448237
				```
			</script>
		</section>

		<section id="taxonomic-ids">
			<h1>Taxonomic IDs</h1>
			<br>
			<h2>always try to move from:</h2>
			<br>
			<ul class="b">
				<li><h3 align="left">taxonomic name -- to</h3></li>
				<li><h3 align="left">taxonomic ID -- to</h3></li>
				<li><h3 align="left">whatever other data</h3></li>
			</ul>
		</section>

		<section data-markdown id="rentrez-example">
			<script type="text/template">
				## ENTREZ in R - <a href="https://github.com/ropensci/rentrez">rentrez</a>

				### Retrieve datasets from a particular organism
				<br>
				```r
				library(rentrez)
				x <- entrez_search(db = "popset", term = "Latrodectus katipo[Organism]")
				x_summs <- entrez_summary(db = "popset", id = x$ids)
				titles <- extract_from_esummary(x_summs, "title")
				COI_ids <- x$ids[c(2,6)]
				COI <- entrez_fetch(db = "popset", id = COI_ids, rettype = "fasta")
				substr(COI, 1, 500)
				#> [1] ">AY383604.1 Latrodectus katipo isolate 1 tRNA-Leu (trnL) gene, 
				#> partial sequence; and NADH dehydrogenase subunit 1 (ND1) gene, 
				#> partial cds; mitochondrial
				#> ATTACCTAAATTATATCAATAATTATTAATTCAGTTATCCCTATAATCTCTCTCCTAATCAGAGTCTCGT
				#> TTTACACTATCTTAGAACGAAAAATTTTAAGTTATATCCAAATTCGTAAAGGGCCTAATAAAGTGGGTTT
				#> CCTTGGTATTCTCCAACCCTTTAGAGACGCAATTAAACTTTTCAACAAAAATCTTATTAAACCTTTCTCT
				#> TCTAATTTTTTAATTTTCTACATCTCCCCCGCCTTATCTTTAACTCTTGCTCTCCTTATTTCTTCAATCA
				#> TTCCTTTCTTTTACTTTTCTCCTTATGATAACAAACATAATATCTTGATATTCTTAATCTTAT"
				```
			</script>
		</section>


		<!-- species occurrence tools -->
		<section id="occurrence-tools">
			<h1>Occurrence data</h1>
			<ul class="b">
				<li><h3 style="font-size: 40px"><a href="https://github.com/ropensci/rgbif">rgbif</a> - GBIF</h3></li>
				<li><h3 style="font-size: 40px"><a href="https://github.com/ropensci/rbison">rbison</a> - USGS's BISON</h3></li>
				<li><h3 style="font-size: 40px"><a href="https://github.com/ropensci/rvertnet">rvertnet</a> - VertNet</h3></li>
				<li><h3 style="font-size: 40px"><a href="https://github.com/ropensci/rebird">rebird</a> - eBird (see also <a href="https://github.com/CornellLabofOrnithology/auk">auk</a>)</h3></li>
				<li><h3 style="font-size: 40px"><a href="https://github.com/ropensci/rinat">rinat</a> - iNaturalist</h3></li>
				<li><h3 style="font-size: 40px"><a href="https://github.com/ropensci/spocc">spocc</a> - one stop shop (of all above)</h3></li>
				<li><h3 style="font-size: 40px"><a href="https://github.com/ropensci/finch">finch</a> - parse GBIF bulk data</h3></li>
				<ul>
					<li><h3 style="font-size: 40px"><a href="https://cran.rstudio.com/web/packages/EML">EML</a> - read and create EML</h3></li>
				</ul>
				<li><h3 style="font-size: 40px"><a href="https://github.com/iDigBio/ridigbio">ridigbio</a>&nbsp;<a href="https://github.com/AtlasOfLivingAustralia/ALA4R">ALA4R</a>&nbsp;<a href="https://github.com/iobis/robis">robis</a> *</h3></li>
			</ul>
		</section>

		<section data-markdown id="rgbif-example">
			<script type="text/template">
				## GBIF data - <a href="https://github.com/ropensci/rgbif">rgbif</a>
				<br>
				### Taxonomic hierarchies from NCBI/ITIS/COL/etc

				```r
				library('rgbif')
				FIXME: add example
				```
			</script>
		</section>

		<!-- tools in development and other unsolved problems -->
		<!-- FIXME: make slides -->




		<!-- talk to us -->
		<section id="talk-to-us" class="center">
			<h1>talk to us <i class="fa fa-bullhorn"></i></h1>
			<br>
			<h2>what would you like to see?</h2>
			<h2>what open data is too hard to get?</h2>
			<h2>discussion forum: <a href="https://discuss.ropensci.org/">discuss.ropensci.org</a></h2>
			<h2>submit a package/review a package: <a href="https://github.com/ropensci/onboarding">github.com/ropensci/onboarding</a></h2>
		</section>

		<!-- end -->
		<section id="end">
			<i class="fa fa-external-link fa-5x"></i>
			<br>
			<h2><a href="https://scotttalks.info/bocc">scotttalks.info/bocc</a></h2>
			<br><br>
			<h2>Made w/: <a href="https://github.com/hakimel/reveal.js">reveal.js v3.2.0</a></h2>
			<br>
			<h2>Some Styling: <a href="http://getbootstrap.com/">Bootstrap v3.3.5</a></h2>
			<br>
			<h2>Icons by: <a href="http://fortawesome.github.io/Font-Awesome/">FontAwesome v4.4.0</a></h2>
		</section>

			</div>

		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.js"></script>

		<script>

			// Full list of configuration options available at:
			// https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				controls: true,
				progress: true,
				history: true,
				center: false,
				slideNumber: true,
				keyboard: true,
				touch: true,

				width: 1400,
				height: 900,

				// transition: 'slide', // none/fade/slide/convex/concave/zoom
				transition: Reveal.getQueryHash().transition || 'none', // default/cube/page/concave/zoom/linear/fade/none

				// Optional reveal.js plugins
				dependencies: [
					{ src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
					{ src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
					{ src: 'plugin/zoom-js/zoom.js', async: true },
					{ src: 'plugin/notes/notes.js', async: true }
				]
			});

		</script>

	</body>
</html>
